cli/src/commands/config/set.rs

```diff
@@ -46,7 +47,85 @@ pub fn cmd_config_set(
             path = config_path.display()
         )));
     }
-    // TODO(#531): Infer types based on schema (w/ --type arg to override).
     let value = parse_toml_value_or_bare_string(&args.value);
+    // If the user is trying to change the author config, we should warn them that
+    // it won't affect the working copy author
+    if args.name == ConfigNamePathBuf::from_iter(vec!["user", "name"]) {
+        check_wc_user_name(command, ui, &value)?;
+    } else if args.name == ConfigNamePathBuf::from_iter(vec!["user", "email"]) {
+        check_wc_user_email(command, ui, &value)?;
```

Comment:
nit: perhaps, `check_wc_user_name`/`email` can be combined as a function that returns `Option<Signature>` or `bool`?

Something like
```rust
    if let Some(author) = get_wc_author(command) {
        if value.as_str() != Some(&author.name)) {
            warn_wc_author(ui, &author)?;
        }
    }
```

Reply:
I combined the functions in the best way I could.